src/metrics/promotions.ts

```diff
@@ -1,54 +1,57 @@
-import { Chess } from '../../cjsmin/src/chess';
-import { FileReaderGame } from '../types';
-
-export async function getPiecePromotionInfo(games: FileReaderGame[]) {
-  console.time('Task 7: getPiecePromotionInfo');
-  let ambigPiecePromotedToMap = {};
-  let promotingPieceMap = {};
-
-  for (const game of games) {
-    const chess = new Chess();
-    chess.loadPgn(game.moves);
-    const chessHistory = chess.history();
+import {
+  Pawns,
+  Piece,
+  PrettyMove,
+  PromotablePiece,
+} from '../../cjsmin/src/chess';
+import { Metric } from './metric';
+
+export class PromotionMetric implements Metric {
+  //uas pawns mapped to count of promotions to each piece type
```

Comment:
3d array for promotionMap? nice. but why necessary?

Reply:
not 3d array. Object with depth 2 mapping keys to promotion count. We want to know what paws promotes to what. And we probably want to know on what square.

Writing the data to a db might be more useful here actually... But we are too deep